### ISSUE TITLE
Мини ML модель, документация, артефакты ONNX

### ISSUE THREAD

**Issue by sidimekov** [sidimekov/asset-price-predict#60]
# Мини модель, документация, артефакты ONNX

## Цель

Подготовить минимальную модель прогноза для клиентского инференса, описать процесс обучения и экспорт в onnx, положить артефакты в репозиторий, чтобы их использовал ML Worker

- простой baseline под браузерный инференс через onnxruntime web
- вход фич одной длины, выход p50 плюс опционально p10 p90
- полная документация в docs, чтобы можно было воспроизвести

Файлы и структура

```
docs/
  modeling/
    forecast_minimal.ipynb                  # (если будет) ноутбук обучение, валидация, экспорт
    forecast_minimal.md                     # (опционально) текстовая версия, краткая инструкция
    data_readme.md                          # (опционально) откуда данные, как скачать, очистка
    test_vectors.json                       # эталонные входы и ожидаемые выходы
apps/web/public/models/
  forecast_minimal.onnx                     # основная модель
  forecast_minimal.onnx.sha256              # контрольная сумма
  forecast_minimal.quant.onnx               # (если необходимо) при наличии квантованная версия
  forecast_minimal.quant.onnx.sha256
apps/web/src/config/ml.ts                   # версия модели, форма входа, нормализация
```

## Требования к модели

- формат onnx opset 17 или ниже для совместимости с onnxruntime web
- размер файла до 5 мб, целевая первая инициализация до 3 с, повторный прогон до 1 с
- вход `X` -> `float32[1, n]` где `n` фиксировано (в контрактах константы горизонта прогноза)
- выход один из вариантов

  - три выхода `p10`, `p50`, `p90` -> `float32[H]` где `H` горизонт
  - один выход `delta` -> `float32[H]`, постпроцесс восстанавливает цены из последнего `close`

- нормализация входа и денормализация выхода описаны в `ml.ts` и в ноутбуке
- совместимость с `feature pipeline` в pr про ml worker, список фич и порядок зафиксированы

## Данные и подготовка

- источник таймсерий для обучения описан в `data_readme.md`
- базовая очистка данные без пропусков по выбранному tf, удаление дубликатов, выравнивание по времени
- train split по времени, валидация по последнему блоку, без утечки будущего

## Базовая архитектура

- один из вариантов

  - Linear Regression с расширенными признаками лаги momentum ema vol календарь
  - LightGBM Quantile регрессия с тремя обучениями p10 p50 p90

- выбор делается по простоте экспорта в onnx
- если выбран LightGBM то экспорт через `skl2onnx` или `onnxmltools`, в ноутбуке зафиксировать шаги
- при невозможности прямого экспорта используем эквивалент в sklearn который точно экспортируется

## Экспорт в ONNX

- ноутбук показывает шаги
  1 подготовка `X`, `y`
  2 обучение
  3 фиксация порядка фич и mean std для нормализации
  4 конвертация в onnx, задание имен входа `input` и выходов `p50` `p10` `p90` или `delta` (перцентили на этом этапе пока что опционально)
  5 проверка `onnxruntime` на CPU локально, сопоставление с исходной моделью на 5 тестовых кейсах
- сохраняем `forecast_minimal.onnx`, считаем sha256, пишем в файл рядом

## Квантование опционально

- post training static quantization если размер больше 5 мб
- проверка регрессии качества на валидации
- если просадка больше порога 5 процентов то оставляем не квантованную версию

## Совместимость с фронтом

- `apps/web/src/config/ml.ts` обновляется

  - `modelVer` -> строка например `min-0-1-0`
  - `inputShape` -> `[1, n]`
  - `horizonSteps` -> число H
  - `normalization` -> тип `zscore` или `robust` и параметры
  - `outputs` -> `['p50']` или `['p10','p50','p90']` или `['delta']`

- в `test_vectors.json` кладем минимум 2 кейса

  - `tail` -> массив `[ [ts, close] ... ]`
  - `expected` -> массивы выходов до округления
  - допустимое отклонение например `rtol 1e-3`

## Документация

- `forecast_minimal.md` кратко если есть что написать

  - что за модель, как запустить ноутбук, как экспортировать, где лежат артефакты
  - как обновить sha256 и `ml.ts`
  - можно написать кратко назначение, данные, метрики MAPE на валидации, ограничения, риски и этические аспекты
  - заметка что модель не инвестиционная рекомендация

- если необходимо указать лицензии источников данных, ссылки

## CI проверки минимальные

- шаг в unit workflow который парсит `ml.ts` и проверяет что указанные файлы onnx существуют и совпадает sha256
- smoke локально через скрипт в ноутбуке или маленький js скрипт вне CI по возможности, CI достаточно валидировать хеш и `test_vectors.json` формат

## DoD

- в репозитории есть `forecast_minimal.onnx` с контрольной суммой, размер до 5 мб
- ноутбук и `forecast_minimal.md` позволяют воспроизвести обучение и экспорт
- `ml.ts` содержит корректные параметры модели и ссылку на файл onnx
- `test_vectors.json` присутствует и понятен, применяется в ml worker (если написан) для e2e
- фронт собирается, модель доступна по `public/models` пути
- команда проекта понимает как обновлять модель и версию
